src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -475,13 +474,25 @@ private static void UpdateGasUp(long refund, ref long gasAvailable)
         gasAvailable += refund;
     }
 
-    private bool ChargeAccountAccessGas(ref long gasAvailable, EvmState vmState, Address address, IReleaseSpec spec, bool chargeForWarm = true)
+    private bool ChargeAccountAccessGas(ref long gasAvailable, EvmState vmState, Address address, bool chargeForDelegation, IReleaseSpec spec, bool chargeForWarm = true)
```

Comment:
I see that we chargeForDelegation for all instructions besides SELFDESTRUCT and BALANCE. Are we sure it is desired and not a mistake in the spec?

Reply:
We charge twice for account access, when it is necessary to lookup the extra address for some OP codes. In the case of SELFDESTRUCT and BALANCE, the delegation is not accessed.